testing/adios2/engine/bp/TestBPWriteRead.cpp

```diff
@@ -102,83 +151,105 @@ TEST_F(BPWriteReadTest, ADIOS2BPWriteADIOS1Read1D8)
         engine->Close();
     }
 
-// Read test data using ADIOS1
-#ifdef ADIOS2_HAVE_MPI
-    // Read everything from rank 0
-    int rank;
-    MPI_Comm_rank();
-    if (rank == 0)
-#endif
     {
         adios_read_init_method(ADIOS_READ_METHOD_BP, MPI_COMM_WORLD,
                                "verbose=3");
 
         // Open the file for reading
-        ADIOS_FILE *f =
-            adios_read_open_file((fname + ".dir/" + fname + ".0").c_str(),
-                                 ADIOS_READ_METHOD_BP, MPI_COMM_WORLD);
+        std::string index = std::to_string(mpiRank);
+        ADIOS_FILE *f = adios_read_open_file(
+            (fname + ".dir/" + fname + "." + index).c_str(),
+            ADIOS_READ_METHOD_BP, MPI_COMM_WORLD);
```

Comment:
@williamfgc @pnorber ADIOS1 can process the parallel files from ADIOS2 right?  And if that's the case, ADIOS1 is reading back each rank's file independently, not in parallel.  In which case, should we initialize ADIOS1 with MPI_COMM_SELF instead?

Reply:
This test is about asking multiple processes write data into a 1D array using `BPFileWriter`. Ex. process 0 would write index 0\~5, process 1 would write index 6\~9 etc.. The result would be a 1D array with size as mpiSize*5.

When reading the data back, if I'm using`MPI_COMM_SELF` then the bounding box selection for each process would be [mpiRank * 5, 5](start and count). With `MPI_COMM_WORLD` the bounding box selection would be [0,5]. Here I'm really confused... Why `MPI_COMM_SELF` would need a global starting index while `MPI_COMM_WORLD` would use a local starting index?